r/R/dplyr-filter.R

```diff
@@ -71,8 +71,15 @@ filter.Dataset <- filter.ArrowTabular <- filter.arrow_dplyr_query
 
 set_filters <- function(.data, expressions) {
   if (length(expressions)) {
-    # expressions is a list of Expressions. AND them together and set them on .data
-    new_filter <- Reduce("&", expressions)
+    if (is.list(expressions)) {
+      # expressions is a list of Expressions. AND them together and set them on .data
+      new_filter <- Reduce("&", expressions)
+    } else {
+      # expressions should be an expression or list of expressions already
+      stopifnot(is_list_of(expressions, "Expression") | inherits(expressions, "Expression"))
+      new_filter <- expressions
+    }
```

Comment:
how do you have a list of expressions here? won't `is.list(expressions)` catch that? do you mean something like this? 

```suggestion
    } else if (inherits(expressions, "Expression")) {
      new_filter <- expressions
    } else {
      stop("Some error message here", call. = FALSE)
    }
```

Reply:
Yeah, this is a holdover from a previous construction where this was outside of the if. Lemme move this around like you suggest.